oremda/pipeline.py

```diff
@@ -154,6 +160,8 @@ def run(self):
         )
         run_operators = set()
 
+        self.observer.on_start(self)
```

Comment:
Is this observer always associated with this pipeline? If that's the case, maybe you should pass in `self` in the `PipelineObserver.__init__()` method and store it inside, so you don't have to pass in `self` for all of the observer functions.

Reply:
I'm not quite set on the API of the PipelineObserver, but the reasoning is that we could use a single instance of the observer to observe multiple pipelines (for example, all the pipelines associated with a single client/session)